### PR TITLE
[runtime] Add a memory manager for dynamic methods, use it to allocate some of the memory allocated by the runtime for dynamic methods.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
The test failures look relevant.

File: mono/metadata/loader-internals.h [hunk 1]
```diff
@@ -89,6 +89,8 @@ struct _MonoMemoryManager {
 	gboolean is_generic;
 	// Whether the MemoryManager is in the process of being freed
 	gboolean freeing;
+	// Whenever this is owned by a dynamic method
+	gboolean dynamic_method;
```
Comment (lambdageek) [inline mono/metadata/loader-internals.h:93]: I think you're adding a third kind of memory manager here.  So maybe it's better to do something like: 

```c
enum  { MEM_MGR_SINGLE, MEM_MGR_GENERIC, MEM_MGR_DYNAMIC_METHOD } MonoMemoryManagerKind;

struct _MonoMemoryManager {
  MonoMemoryManagerKind kind;
   ...
};

struct _MonoMemoryManagerDynamicMethod {
   MonoMemoryManager  memory_manager;
};
```

**Review by lambdageek — COMMENTED** [review]
Maybe it's better to keep "MonoMemoryManager" as an abstract base class?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
Closing until issues with it are fixed.

**Pull request closed without merging**